sesman/sesman.c

```diff
@@ -187,6 +246,90 @@ static int sesman_listen_test(struct config_sesman *cfg)
 
     return rv;
 }
+
+/******************************************************************************/
+int
+sesman_close_all(void)
+{
+    int index;
+    struct sesman_con *sc;
+
+    LOG_DEVEL(LOG_LEVEL_TRACE, "sesman_close_all:");
+    trans_delete(g_list_trans);
+    for (index = 0; index < g_con_list->count; index++)
+    {
+        sc = (struct sesman_con *) list_get_item(g_con_list, index);
+        delete_connection(sc);
+    }
+    return 0;
+}
+
+/******************************************************************************/
+static int
+sesman_data_in(struct trans *self)
+{
+    int version;
+    int size;
+
+    if (self->extra_flags == 0)
+    {
+        in_uint32_be(self->in_s, version);
+        in_uint32_be(self->in_s, size);
+        if (size > self->in_s->size)
+        {
+            LOG(LOG_LEVEL_ERROR, "sesman_data_in: bad message size");
+            return 1;
+        }
+        self->header_size = size;
+        self->extra_flags = 1;
+    }
+    else
+    {
+        /* process message */
+        struct sesman_con *sc = (struct sesman_con *)self->callback_data;
+        self->in_s->p = self->in_s->data;
+        if (scp_process(self, sc->s) != SCP_SERVER_STATE_OK)
+        {
+            LOG(LOG_LEVEL_ERROR, "sesman_data_in: scp_process_msg failed");
+            return 1;
+        }
+        /* reset for next message */
+        self->header_size = 8;
+        self->extra_flags = 0;
+        init_stream(self->in_s, 0);
```

Comment:
setting the transport input stream to size zero doesn't feel right. Can you please clarify this?

Reply:
I'll add a comment. The effect is to reset the stream pointers, but leave the buffer size unchanged.